### PR TITLE
samples: nrf52: replace power_mgr with system_off

### PR DESCRIPTION
**Pull request by pabigot**
The Nordic SOC doesn't support multiple system power states, only one
deep sleep state.  Replace the old example with a simpler one that can
be used to measure CPU active, normal sleep, and system off modes.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

### CODE REVIEW

**Review by mnkp — APPROVED** [review]

File: samples/boards/nrf52/system_off/README.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+nRF5x System Off demo
+#####################
+
+Overview
+********
+
+This sample can be used for basic power measurement and as an example of
+deep sleep on Nordic platforms.  The functional behavior is:
+* Busy-wait for 1 second
```
Comment (dbkinder) [inline samples/boards/nrf52/system_off/README.rst]: Add a blank line before the first item in a list

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/boards/nrf52/system_off/README.rst (hunk 1 above)
Comment (pabigot) [reply samples/boards/nrf52/system_off/README.rst]: done

File: samples/boards/nrf52/power_mgr/src/power.c [hunk 2]
```diff
@@ -1,95 +0,0 @@
...
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#include "sample.h"
-
-/*
- * Application defined function for doing any target specific operations
- * for low power entry.
- */
-void sys_pm_notify_power_state_entry(enum power_states state)
```
Comment (jfischer-phytec-iot) [inline samples/boards/nrf52/power_mgr/src/power.c]: There is no other example how to use entry/exit hooks, but they can be useful for example to set dcdc converter mode.

Comment (pabigot) [reply samples/boards/nrf52/power_mgr/src/power.c]: That's a fair point, but it should probably be a separate example with explanatory context.  I'm not clear on why I'd want to do that through power state events.

**Comment by pabigot** [comment]
@carlescufi and others: please see #20792.  I'm moving towards an opinion the sys_pm API is a little too dangerous to be exposing as an example.

I tried switching to `sys_pm_force_power_state(SYS_POWER_STATE_DEEP_SLEEP_1)` in the hopes it could be used by disabling the state for non-forced entry, but that returns immediately without doing anything, and the API doesn't allow for an error value.

**Comment by pabigot** [comment]
Updated with revised code that seems to be safer, though it doesn't do a system power-off when you fall off the end of main.

**Review by carlescufi — APPROVED** [review]
LGTM, thanks!

File: samples/boards/nrf52/system_off/README.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#####################
+
+Overview
+********
+
+This sample can be used for basic power measurement and as an example of
+deep sleep on Nordic platforms.  The functional behavior is:
+
+* Busy-wait for 1 second
+* Sleep for 1 second
```
Comment (anangl) [inline samples/boards/nrf52/system_off/README.rst]: Both the above times are actually 2 seconds.

**Review by anangl — APPROVED** [review]
LGTM

File: samples/boards/nrf52/system_off/README.rst (hunk 3 above)
Comment (pabigot) [reply samples/boards/nrf52/system_off/README.rst]: Thanks; corrected

**Comment by pabigot** [comment]
@dbkinder Please revisit this; your change request should be complete.  Thanks.

**Review by dbkinder — APPROVED** [review]
docs LGTM, thanks!

**Comment by pabigot** [comment]
Inadvertently closed while attempting to fix result of merging #21135.

**Pull request merged**